installation/centos-setup.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+#  This file is part of cc-runtime.
+#
+#  Copyright (C) 2017 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# Description: This script installs Clear Containers on a
+#   CentOS 7 system.
+#
+
+# all errors are fatal
+set -e
+
+# enable tracing
+set -x
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/installation-setup.sh"
+source "${SCRIPT_PATH}/../versions.txt"
+
+# List of packages to install to satisfy build dependencies
+pkgs=""
+
+mnl_dev_pkg="libmnl-devel"
+
+# general
+pkgs+=" zlib-devel"
+pkgs+=" gettext-devel"
+pkgs+=" libtool-ltdl-devel"
+pkgs+=" libtool-ltdl"
+pkgs+=" glib2-devel"
+pkgs+=" bzip2"
+pkgs+=" m4"
+
+# for yum-config-manager
+pkgs+=" yum-utils"
+
+# runtime dependencies
+pkgs+=" libuuid-devel"
+pkgs+=" libmnl"
+pkgs+=" ${mnl_dev_pkg}"
+pkgs+=" libffi-devel"
+pkgs+=" pcre-devel"
+
+# qemu lite dependencies
+pkgs+=" libattr-devel"
+pkgs+=" libcap-devel"
+pkgs+=" libcap-ng-devel"
+pkgs+=" pixman-devel"
+
+pkgs+=" gcc-c++"
+
+source /etc/os-release
+
+major_version=$(echo "${VERSION_ID}"|cut -d\. -f1)
+
+if [ "${os_distribution}" = centos ]
+then
+    distro="CentOS"
+else
+    echo >&2 "ERROR: Unrecognised distribution: ${os_distribution}"
+    echo >&2 "ERROR: This script is designed to work on CentOS systems only."
+    exit 1
+fi
+
+site="http://download.opensuse.org"
+dir="repositories/home:/clearcontainers:/clear-containers-3/${distro}_${major_version}"
+repo_file="home:clearcontainers:clear-containers-3.repo"
+cc_repo_url="${site}/${dir}/${repo_file}"
+
+sudo yum -y update
+eval sudo yum -y install "${pkgs}"
+
+sudo yum groupinstall -y 'Development Tools'
+
+pushd "${deps_dir}"
+
+# Install pre-requisites for gcc
+gmp_file="gmp-${gmp_version}.tar.bz2"
+curl -L -O "ftp://gcc.gnu.org/pub/gcc/infrastructure/${gmp_file}"
```

Comment:
Could we try to use always http/https if possible (some times user might be behind proxy and it might not work 100% with ftp)